packages/flutter/lib/src/widgets/app.dart

```diff
@@ -413,6 +415,7 @@ class WidgetsApp extends StatefulWidget {
   /// {@endtemplate}
   WidgetsApp.router({
     super.key,
+    this.routeInformationCodec,
```

Comment:
Instead of wiring through a separate property just for the codec, could the codec semantically live on one of the existing router-related objects you pass in? Or does it not make sense to bundle it with any of those?

Reply:
yes I think that can be done.

Reply:
Which one would be the best candidate?